src/datasets/nllb_dataset.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import torch
+from datasets import load_dataset
+from src.utils.common import clean_text
+from src.utils.slicer import VTRSlicer
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import IterableDataset
+from tqdm import tqdm
+
+
+def collate_batch_common(slices: list[torch.Tensor], labels: list[int]):
+    # [batch size; most slices; font size; window size]
+    batched_slices = pad_sequence(slices, batch_first=True, padding_value=0.0).float()
+    bs, ms, _, _ = batched_slices.shape
+
+    # [batch size; most slices]
+    attention_mask = torch.zeros((bs, ms), dtype=torch.float)
+    for i, s in enumerate(slices):
+        attention_mask[i, : len(s)] = 1
+
+    return {
+        "slices": batched_slices,
+        "attention_mask": attention_mask,
+        "labels": torch.tensor(labels, dtype=torch.float),
+    }
+
+
+class DatasetNLLB(IterableDataset):
```

Comment:
Let's also support the translation task. We will need it soon.
Let `DatasetNLLB` return a complete pair: both sentences and languages.
After that, let's create a `DatasetNLLBLId` that inherits from `DatasetNLLB`. And it overrides the `__iter__` method, it calls `super` and yields each sentence with language separately.